framework/include/mediaplayer/mediaplayer.h

```diff
@@ -0,0 +1,215 @@
+#ifndef __MEDIAPLAYER_H__
+#define __MEDIAPLAYER_H__
+
+#include <queue.h>
+#include <sys/types.h>
+
+/***
+ * Errors
+ */
+#define MEDIA_OK					0
+#define MEDIA_ERROR					1
+#define MEDIA_ERROR_THREAD_INIT		2
+#define MEDIA_ERROR_PCM_WRITE		3
+#define MEDIA_ERROR_WRITE_TO_FILE	4
+#define MEDIA_ERROR_CREATE			5
+#define MEDIA_ERROR_UNKNOWN_FILE	6
+
+/***
+ * Enumerations
+ */
+typedef enum {
+	MEDIA_TYPE_PCM,
+	MEDIA_TYPE_PCM_STREAM,
+	MEDIA_TYPE_WAV,
+	MEDIA_TYPE_WAV_STREAM
+} media_type_e;
+
+typedef enum {
+	MEDIA_OP_RECORD,
+	MEDIA_OP_PLAYBACK
+} media_op_e;
+
+typedef enum {
+	MEDIA_STATE_CREATED,
```

Comment:
I think play & record can use same state...
MEDIA_STATE_READY -> ready to play
MEDIA_STATE_RUNNING -> play or record
MEDIA_STATE_PAUSE -> pause
MEDIA_STATE_STOP -> stop
MEDIA_STATE_CLOSE(TERMINATE??) -> finished

Reply:
Yes both play and record share these enum definitions

Reply:
I mean we don't have to distinguish play & record.

Reply:
media_open needs this flag to classify the media operation type. It sends all records operation to the record thread and all playback operations to playback thread.